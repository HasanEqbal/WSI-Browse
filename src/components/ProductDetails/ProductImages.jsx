import React, { useState } from 'react';
import './ProductImages.scss';
import Overlay from '../Overlay/Overlay'

export default function ProductsImages({ product }) {
    const [mainImage, setMainImage] = useState(product[0].hero)
    const [displayCarosuel, setDisplayCarosuel] = useState(false)
    let { images } = product[0]
    return (
        <React.Fragment>
            {<div className="medium-6 columns">
                <a>
                    <img className="thumbnail" onClick={() => { setDisplayCarosuel(true) }} src={mainImage} />
                </a>
                <div className="row small-up-8">
                    {images ? images.map((image, index) => {
                        return (<div key={index} className="column">
                            <a>
                                <img className="thumbnail" src={image.href} onClick={(e) => { setMainImage(e.target.src) }} />
                            </a>
                        </div>)
                    }) : <h4>Product Does not have alternative images</h4>}
                </div>
            </div>}
            {displayCarosuel && <Overlay images={images} />}
        </React.Fragment >
    )
}


