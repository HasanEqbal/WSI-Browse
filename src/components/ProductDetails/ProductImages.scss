.small-up-8 {
  margin-top: 10px
}
p{
  font-size: 14px;
}
.small-up-8 .column{
  padding-left: 0px
}
