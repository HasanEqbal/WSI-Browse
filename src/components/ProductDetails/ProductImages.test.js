import React from 'react';
import { shallow } from 'enzyme';
import ProductImages from './ProductImages';
const fixture = require('../../testData/testFixture.json');

describe('Component : ProductImages', () => {

  it('expect to render Product Images component', () => {
    const product = fixture.groups[0]
    expect(shallow(<ProductImages product={product} />)).toMatchSnapshot();
  })

  it('verify the clicking of alternative images should display as main image', () => {
    const product = fixture.groups[0]
    const event = {
      target: {
        src: 'some-url'
      }
    }
    const wrapper = shallow(<ProductImages product={product} />)
    wrapper.find('img.alt-image').at(0).simulate('click', event)
    expect(wrapper.find('img.main-image').at(0).props().src) === event.target.src
  })

  it('verify the clicking on main image overlay should display', () => {
    const product = fixture.groups[0]
    const event = {
      target: {
        src: 'some-url'
      }
    }
    const wrapper = shallow(<ProductImages product={product} />)
    wrapper.find('img.main-image').at(0).simulate('click', event)
    expect(wrapper.find('div.overlay')) === true
  })

})
