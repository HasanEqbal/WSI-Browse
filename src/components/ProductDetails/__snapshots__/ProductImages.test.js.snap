// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component : ProductImages expect to render Product Images component 1`] = `ShallowWrapper {}`;
